### PR TITLE
feat(1610): Add cleanup method

### PR DESCRIPTION
**Pull request by pritamstyz4ever**
## Context

Currently when the node process terminates it happens abruptly and the in process messages are lost

## Objective

This PR adds an abstract cleanup function

## References

https://github.com/screwdriver-cd/screwdriver/issues/1610

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -157,6 +157,18 @@ class Executor {
...
+    /**
+     * Clean up any processing tasks
+     * @method cleanUp
+     */
+    async cleanUp() {
+        await this._cleanUp();
+    }
+
+    async _cleanUp() {
+        // no-op in case no implemented in extenders
```
Comment (DekusDenial) [inline index.js]: is it proper to enforce mandatory clean up?

Comment (pritamstyz4ever) [reply index.js]: Yes, tried to keep changes to minimal

**Comment by pritamstyz4ever** [comment]
> Do you need to add `_cleanUp` method to `executor-router` as well?
> https://github.com/screwdriver-cd/executor-router/blob/master/index.js

executor-router seems to be called from queue-worker, the cleanup is only restricted to the queue for now.

**Comment by tkyi** [comment]
Can you update README?

**Pull request merged**